js/private/js_run_devserver.mjs

```diff
@@ -358,15 +359,18 @@ async function syncFiles(files, sandbox, writePerm) {
         )
     }
 
-    totalSynced += (
-        await Promise.all(
-            otherNodeModulesFiles.map(async (file) => {
-                const src = path.join(RUNFILES_ROOT, file)
-                const dst = path.join(sandbox, file)
-                return await syncRecursive(src, dst, sandbox, writePerm)
-            })
-        )
-    ).reduce((s, t) => s + t, 0)
+    syncPromises = syncPromises.concat(
+        otherNodeModulesFiles.map((file) => {
+            const src = path.join(RUNFILES_ROOT, file)
+            const dst = path.join(sandbox, file)
+            return syncRecursive(src, dst, sandbox, writePerm)
+        })
+    )
+
+    const totalSynced = (await Promise.all(syncPromises)).reduce(
```

Comment:
Doing all of these in parallel could cause problems since Node.js doesn't like symlinks being created to files that don't exist. I think we need to `Promise.all` three times for the three tronches.
1)
```
    // Sync non-node_modules files first since syncing 1p js_library linked node_modules symlinks
    // requires the files they point to be in place.
```
2)
```
    // Sync first-party package store files before other node_modules files since correctly syncing
    // direct 1p node_modules symlinks depends on checking if the package store synced files exist.
```
3)
```
// Finally sync all remaining node_modules files
```

Reply:
I've dropped this change for now.

Reply:
👍